pandas/io/common.py

```diff
@@ -201,10 +201,22 @@ def get_filepath_or_buffer(
         if filepath_or_buffer.startswith("s3n://"):
             filepath_or_buffer = filepath_or_buffer.replace("s3n://", "s3://")
         fsspec = import_optional_dependency("fsspec")
+        from botocore.exceptions import NoCredentialsError
```

Comment:
It's not necessarily sure that botocore is installed at this point? (there are other fsspec filesystems that can end up here and not only s3fs)

Reply:
Yeah updated - thoughts on this approach?